docker-compose issues

When running the docker-compose file in the readme, the rubrix_1 service fails with the following exception stack trace:

```
[2021-05-04 16:23:21 +0000] [40] [ERROR] Traceback (most recent call last):
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
rubrix_1         |     (self._dns_host, self.port), self.timeout, **extra_kw
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
rubrix_1         |     raise err
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
rubrix_1         |     sock.connect(sa)
rubrix_1         | ConnectionRefusedError: [Errno 111] Connection refused
rubrix_1         | 
rubrix_1         | During handling of the above exception, another exception occurred:
rubrix_1         | 
rubrix_1         | Traceback (most recent call last):
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 252, in perform_request
rubrix_1         |     method, url, body, retries=Retry(False), headers=request_headers, **kw
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
rubrix_1         |     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 507, in increment
rubrix_1         |     raise six.reraise(type(error), error, _stacktrace)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 735, in reraise
rubrix_1         |     raise value
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
rubrix_1         |     chunked=chunked,
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
rubrix_1         |     conn.request(method, url, **httplib_request_kw)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
rubrix_1         |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
rubrix_1         |   File "/usr/local/lib/python3.7/http/client.py", line 1252, in request
rubrix_1         |     self._send_request(method, url, body, headers, encode_chunked)
rubrix_1         |   File "/usr/local/lib/python3.7/http/client.py", line 1298, in _send_request
rubrix_1         |     self.endheaders(body, encode_chunked=encode_chunked)
rubrix_1         |   File "/usr/local/lib/python3.7/http/client.py", line 1247, in endheaders
rubrix_1         |     self._send_output(message_body, encode_chunked=encode_chunked)
rubrix_1         |   File "/usr/local/lib/python3.7/http/client.py", line 1026, in _send_output
rubrix_1         |     self.send(msg)
rubrix_1         |   File "/usr/local/lib/python3.7/http/client.py", line 966, in send
rubrix_1         |     self.connect()
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
rubrix_1         |     conn = self._new_conn()
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
rubrix_1         |     self, "Failed to establish a new connection: %s" % e
rubrix_1         | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fd72c079b50>: Failed to establish a new connection: [Errno 111] Connection refused
rubrix_1         | 
rubrix_1         | During handling of the above exception, another exception occurred:
rubrix_1         | 
rubrix_1         | Traceback (most recent call last):
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 526, in lifespan
rubrix_1         |     async for item in self.lifespan_context(app):
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 467, in default_lifespan
rubrix_1         |     await self.startup()
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 502, in startup
rubrix_1         |     await handler()
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/rubrix/server/server.py", line 71, in configure_elasticsearch
rubrix_1         |     datasets: DatasetsDAO = create_datasets_dao(es=es_wrapper)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/rubrix/server/datasets/dao.py", line 271, in create_datasets_dao
rubrix_1         |     _instance = DatasetsDAO(es)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/rubrix/server/datasets/dao.py", line 66, in __init__
rubrix_1         |     self.init()
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/rubrix/server/datasets/dao.py", line 73, in init
rubrix_1         |     force_recreate=True,
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/rubrix/server/commons/es_wrapper.py", line 140, in create_index_template
rubrix_1         |     self.__client__.indices.put_template(name=name, body=template)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/client/utils.py", line 153, in _wrapped
rubrix_1         |     return func(*args, params=params, headers=headers, **kwargs)
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/client/indices.py", line 647, in put_template
rubrix_1         |     body=body,
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 413, in perform_request
rubrix_1         |     raise e
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 388, in perform_request
rubrix_1         |     timeout=timeout,
rubrix_1         |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 264, in perform_request
rubrix_1         |     raise ConnectionError("N/A", str(e), e)
rubrix_1         | elasticsearch.exceptions.ConnectionError: ConnectionError(<urllib3.connection.HTTPConnection object at 0x7fd72c079b50>: Failed to establish a new connection: [Errno 111] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x7fd72c079b50>: Failed to establish a new connection: [Errno 111] Connection refused)
rubrix_1         | 
rubrix_1         | [2021-05-04 16:23:21 +0000] [40] [ERROR] Application startup failed. Exiting.
```

--- a/dev.Dockerfile
+++ b/dev.Dockerfile
@@ -23,7 +23,9 @@ RUN git log --oneline \
 
 WORKDIR /app
 
-RUN rm -rf /build
+RUN rm -rf /build \
+ && wget -O /wait-for-it.sh https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh \
+ && chmod +x /wait-for-it.sh
 
 # See <https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker#module_name>
 ENV MODULE_NAME="rubrix.server.server"

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -1,4 +1,4 @@
-version: "3.9"
+version: "3"
 
 services:
   rubrix:
@@ -7,6 +7,8 @@ services:
       - "6900:80"
     environment:
       ELASTICSEARCH: http://elasticsearch:9200
+
+    command: /wait-for-it.sh elasticsearch:9200 -- /start.sh
 
     networks:
       - rubrix

